includes/subscriptions/class-wc-payments-subscriptions-empty-state-manager.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Class WC_Payments_Subscriptions_Empty_State.
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class for loading WooCommerce Payments Subscription empty state screen.
+ */
+class WC_Payments_Subscriptions_Empty_State_Manager {
+
+	use WC_Payments_Subscriptions_Utilities;
+
+	/**
+	 * WC_Payments_Account instance to get information about the account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * WC_Payments_Subscriptions_Empty_State Constructor
+	 *
+	 * @param WC_Payments_Account $account Account class instance.
+	 */
+	public function __construct( WC_Payments_Account $account ) {
+		$this->account = $account;
+
+		if ( ! $this->is_subscriptions_plugin_active() ) {
+			add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts_and_styles' ] );
+			add_filter( 'woocommerce_subscriptions_not_found_label', [ $this, 'replace_subscriptions_empty_state' ] );
+		}
+	}
+
+	/**
+	 * Enqueues the WCPay Subscription empty state scripts and styles.
+	 */
+	public function enqueue_scripts_and_styles() {
+		$screen = get_current_screen();
+
+		// Only enqueue the scripts on the admin subscriptions screen.
+		if ( ! $screen || 'edit-shop_subscription' !== $screen->id ) {
+			return;
+		}
+
```

Comment:
Should also exit early when the store has at least one subscription?

I am seeing the following error on page load: 

<img width="394" alt="Screen Shot 2021-11-23 at 11 37 26 am" src="https://user-images.githubusercontent.com/1527164/142955398-d1e55e67-0e20-4846-8282-b2961ade4ea1.png">

Reply:
Oh, good find. This triggered me to see what happens if you filter the subscriptions table so no results are returned and basically the default "Subscriptions will appear here for..." message is displayed. 

<img width="964" alt="Screen Shot 2021-11-23 at 11 06 53 am" src="https://user-images.githubusercontent.com/8490476/142957341-8354ae4e-8e76-4d50-80ae-165de9b1584b.png">

On this branch, it gets changed to the empty state screen. I'm guessing we don't want it displayed then too?

Reply:
Ah, interesting point about filtering. I wonder if this is something we should tackle in core?

i.e. it's a little odd that we're displaying that empty state when subscriptions exist in the store but you filter by certain criteria that produce 0 results. I would more expect a "0 subscriptions found" type message instead.

Reply:
yeah exactly. I'll open the issues for it.

Reply:
hmm so looking at the code, it looks like the behaviour you described should already be the case https://github.com/automattic/woocommerce-subscriptions-core/blob/1.1.0/includes/class-wc-subscriptions-core-plugin.php#L313-L316. 

I'll double-check I'm not missing something here.

Reply:
I don't suppose it has something to do with cache and the SQL queries you provided in the issue description? 😅

Reply:
yeah it does 🤦‍♂️

Reply:
It's related to the cache. Because we store a [bool in a transient](https://github.com/Automattic/woocommerce-subscriptions-core/blob/0bae09c4e64617a23a6d8706883e0e345f449ad6/includes/class-wcs-cached-data-manager.php#L61) it gets converted into an int. That then fails [this `true === check]`(https://github.com/automattic/woocommerce-subscriptions-core/blob/1.1.0/includes/class-wc-subscriptions-core-plugin.php#L315).

Reply:
Closing the loop. Heres the core issue: https://github.com/Automattic/woocommerce-subscriptions-core/issues/51

Reply:
> I am seeing the following console error on page load

I've fixed the error you mentioned when this code runs on stores with more than 1 subscription in 3a9d812.